clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchFieldMapper.scala

```diff
@@ -20,6 +20,16 @@ private[dataaccess] sealed abstract class ElasticsearchFieldMapper(
   def stringToDefinition(fieldType: Type): String => FieldDefinition
 
   def valueToQuery(fieldName: String, fieldType: Type): Any => QueryDefinition
+
+  protected def is[A: TypeTag](tpe: Type): Boolean = {
```

Comment:
I love the refactor, but this can actually be even cleaner syntactically. By defining: 
```scala
implicit class IsTypeTag(tpe: Type) {
    def is[A: TypeTag]: Boolean = {
      /*
     * Elasticsearch has no notion of an array field; instead, any
     * field can hold 1+ values of the same type.
     */
      tpe =:= typeOf[A] || tpe =:= typeOf[Option[A]] || tpe <:< typeOf[Seq[A]] || tpe <:< typeOf[
        Option[Seq[A]]
        ]
    }
  }
```
checking the type of a `tpe` becomes `tpe.is[Boolean]`, instead of `is[Boolean](tpe)`

Reply:
If we were doing this sort of check all over the place it might be worth the DSL-y-ness, but I'm not sure it's worth doing here...

Reply:
Seem like more or less the same to me, both seem equally readable IMO.